src/main/kotlin/com/exactpro/th2/netty/bytebuf/util/ByteBufDataUtil.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:JvmName("ByteBufDataUtil")
+
+package com.exactpro.th2.netty.bytebuf.util
+
+import io.netty.buffer.ByteBuf
+import io.netty.buffer.ByteBufUtil
+import java.nio.charset.Charset
+import java.util.*
+
+const val DEFAULT_PAD_CHAR = '\u0000'
+
+fun ByteBuf.readAsciiChar(): Char = this.readCharSequence(1, Charsets.US_ASCII)[0]
+
+fun ByteBuf.writeAsciiChar(value: Char): ByteBuf = this.apply {
+    require(value.code < 256) {
+        "The $value char with ${value.code} code is out of ASCII table"
+    }
+    ByteBufUtil.writeAscii(this, value.toString())
+}
+
+@JvmOverloads
+fun ByteBuf.readStringPadEnd(
+    length: Int,
+    padChar: Char = DEFAULT_PAD_CHAR,
+    charset: Charset = Charsets.US_ASCII
+): String = readCharSequence(length, charset).toString().substringBefore(padChar)
```

Comment:
> .substringBefore(padChar)

Won't work properly if string contains `padChar` between other chars.
Maybe also change return type to `CharSequence`? This way we can avoid `toString` conversion.